external/kafka/src/test/scala/org/apache/spark/streaming/kafka/DirectKafkaStreamSuite.scala

```diff
@@ -353,10 +353,52 @@ class DirectKafkaStreamSuite
     ssc.stop()
   }
 
+  test("maxMessagesPerPartition with backpressure disabled") {
+    val topic = "maxMessagesPerPartition"
+    def getRateController(
```

Comment:
`getRateController` is not necessary. You can just pass a rate controller to `getDirectKafkaStream` directly